### PR TITLE
Component: elixir banner

### PR DESCRIPTION
**Pull request by khawkins98**
For  #1572

- [x] finish building the template markup
- [ ] add any special css required

### CODE REVIEW

**Comment by khawkins98** [comment]
This is now ready. 

It's a much-scaled option compared to the 1.x option — that's based on real-world usage, and to serve as an alpha-level reset for future development. 

This leverages vf-banner and vf-flag and currently avoids writing any new CSS.

image

File: components/vf-banner-elixir/vf-banner-elixir.react.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+// We use vanilla JS templates for react for compatibility with create react app
+// ---
+import React from "react";
+import Fragment from "react-dom-fragment";
+// eslint-disable-next-line no-unused-vars
+import VfBanner-ElixirTemplate from "./vf-banner-elixir.precompiled.js"; // import templates before the nunjucks env
+import { vfNunjucksEnv } from "@visual-framework/vf-extensions-react/vf-extensions-react.js";
+
+// any JS actions needed on component insertion
+class VfBanner-ElixirCallback extends React.Component {
```
Comment (nitinja) [inline components/vf-banner-elixir/vf-banner-elixir.react.js]: I don't think anything with `-` is a valid name for a class, it wouldn't compile.

File: components/vf-banner-elixir/vf-banner-elixir.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  * The global function for this component
+  * @example vfBanner-Elixir(scope)
+ * @param {object} [scope] - the html scope to process, optional, defaults to `document`
+  */
+function vfBannerElixir(scope) {
+  /* eslint-disable no-redeclare */
+  var scope = scope || document;
+  /* eslint-enable no-redeclare */
+  const elixirBanner = scope.querySelectorAll("[data-vf-js-banner-elixir]");
+  if (!elixirBanner) {
```
Comment (nitinja) [inline components/vf-banner-elixir/vf-banner-elixir.js]: non-blocking comment: this will never be null, so unnecessary nullcheck

**Review by nitinja — APPROVED** [review]
React component will fail (in my opinion, could be wrong). otherwise everything's ok.

File: components/vf-banner-elixir/vf-banner-elixir.react.js (hunk 1 above)
Comment (khawkins98) [reply components/vf-banner-elixir/vf-banner-elixir.react.js]: Ah, thanks there was a bug in the component generator (since fixed)

Comment (khawkins98) [reply components/vf-banner-elixir/vf-banner-elixir.react.js]: Pushed a fix

**Review by nitinja — APPROVED** [review]

File: components/vf-banner-elixir/vf-banner-elixir.js (hunk 2 above)
Comment (khawkins98) [reply components/vf-banner-elixir/vf-banner-elixir.js]: Thanks. Updated. I've been burned by not including this check before — but in this case, yeah, redundant. good catch

**Pull request merged**